receiver/haproxyreceiver/scraper.go

```diff
@@ -15,22 +15,86 @@
 package haproxyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver"
 
 import (
+	"bytes"
 	"context"
+	"encoding/csv"
+	"net"
+	"strings"
+	"time"
 
-	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver/internal/metadata"
 )
 
 type scraper struct {
+	endpoint       string
+	logger         *zap.Logger
+	metricsBuilder *metadata.MetricsBuilder
 }
 
-func (s scraper) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	return pmetric.NewMetrics(), nil
+func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+
+	c, err := net.Dial("unix", s.endpoint)
```

Comment:
There is a way to make this work with context but it isn't intuitive.

```suggestion
        var d net.Dialer
	c, err := d.DialContext(ctx, "unix", s.endpoint)
```

Reply:
I would also double check that you need `unix` type socket, instead of a dgram socket which has the network `unixgram`

Reply:
I might add a new functionality where we could either use a file endpoint or a tcp endpoint, and I guess I need some real field testing.